src/java.base/share/classes/java/lang/AbstractStringBuilder.java

```diff
@@ -769,16 +775,91 @@ public AbstractStringBuilder append(char c) {
      * @param   i   an {@code int}.
      * @return  a reference to this object.
      */
+    @ForceInline
     public AbstractStringBuilder append(int i) {
         int count = this.count;
-        int spaceNeeded = count + Integer.stringSize(i);
-        ensureCapacityInternal(spaceNeeded);
-        if (isLatin1()) {
-            Integer.getChars(i, spaceNeeded, value);
-        } else {
+
```

Comment:
Note, this method has corresponding implementation in stringopt:
https://github.com/openjdk/jdk/blob/013367b4831094cdd330564378de69deccd0dc4b/src/hotspot/share/opto/stringopts.cpp#L1164

https://github.com/openjdk/jdk/blob/013367b4831094cdd330564378de69deccd0dc4b/src/hotspot/share/opto/stringopts.cpp#L1266

If you make any changes, I think we should add comments or leave follow up issues to further sync between Java and c2 implemention.